libs/extensions/std/exec/src/archive-interpreter-executor.ts

```diff
@@ -55,12 +56,69 @@ export class ArchiveInterpreterExecutor extends AbstractBlockExecutor<
       }
       return R.ok(fs.right);
     }
+    if (archiveType === 'gz') {
+      const fs = await this.loadGzFileToInMemoryFileSystem(
+        archiveFile,
+        context,
+      );
+      if (R.isErr(fs)) {
+        return fs;
+      }
+      return R.ok(fs.right);
+    }
+
     return R.err({
       message: `Archive is not a zip-archive`,
       diagnostic: { node: context.getCurrentNode(), property: 'name' },
     });
   }
 
+  private async loadGzFileToInMemoryFileSystem(
```

Comment:
This function should duplicate a lot of the `loadZipFileToInMemoryFileSystem` logic. I assume only the extraction of files would be different, putting them into our in memory filesystem etc is all repeated logic.

It would be better to implement it that way, e.g., by extracting the shared logic of putting files into our in-memory file system into one function and reduce these two functions to just extracting the relevant archives and then using the new function to put files into a file system.